sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -66,6 +66,10 @@ trait TimestampFormatterHelper extends TimeZoneAwareExpression {
 
   protected def isParsing: Boolean
 
+  // Whether the timestamp formatter is for TimestampWithoutTZType.
+  // If yes, the formatter is always `Iso8601TimestampFormatter`.
```

Comment:
Shall we put this logic in `TimestampFormatterHelper`?
```
final protected def getFormatter(fmt: String): TimestampFormatter = {
  if (forTimestampWithoutTZ) new Iso8601TimestampFormatter(...) else ...
}
```

Reply:
Well I prefer to put all the logic in `TimestampFormatter.getFormatter`